test/e2e/pivoting_test.go

```diff
@@ -24,146 +26,210 @@ import (
 
 	pivotingCluster := framework.DiscoveryAndWaitForCluster(ctx, framework.DiscoveryAndWaitForClusterInput{
-		Getter:    targetCluster.GetClient(),
-		Namespace: namespace,
-		Name:      clusterName,
-	}, e2eConfig.GetIntervals(specName, "wait-cluster")...)
+		Getter:    input.TargetCluster.GetClient(),
+		Namespace: input.Namespace,
+		Name:      input.ClusterName,
+	}, input.E2EConfig.GetIntervals(input.SpecName, "wait-cluster")...)
 
 	controlPlane := framework.GetKubeadmControlPlaneByCluster(ctx, framework.GetKubeadmControlPlaneByClusterInput{
-		Lister:      targetCluster.GetClient(),
+		Lister:      input.TargetCluster.GetClient(),
 		ClusterName: pivotingCluster.Name,
 		Namespace:   pivotingCluster.Namespace,
 	})
 	Expect(controlPlane).ToNot(BeNil())
 
 	By("Check that BMHs are in provisioned state")
 	waitForNumBmhInState(ctx, bmov1alpha1.StateProvisioned, waitForNumInput{
-		Client:    targetCluster.GetClient(),
-		Options:   []client.ListOption{client.InNamespace(namespace)},
+		Client:    input.TargetCluster.GetClient(),
+		Options:   []client.ListOption{client.InNamespace(input.Namespace)},
 		Replicas:  numberOfAllBmh,
-		Intervals: e2eConfig.GetIntervals(specName, "wait-object-provisioned"),
+		Intervals: input.E2EConfig.GetIntervals(input.SpecName, "wait-object-provisioned"),
 	})
 
 	By("Check if metal3machines become ready.")
 	waitForNumMetal3MachinesReady(ctx, waitForNumInput{
-		Client:    targetCluster.GetClient(),
-		Options:   []client.ListOption{client.InNamespace(namespace)},
+		Client:    input.TargetCluster.GetClient(),
+		Options:   []client.ListOption{client.InNamespace(input.Namespace)},
 		Replicas:  numberOfAllBmh,
-		Intervals: e2eConfig.GetIntervals(specName, "wait-object-provisioned"),
+		Intervals: input.E2EConfig.GetIntervals(input.SpecName, "wait-object-provisioned"),
 	})
 
 	By("Check that all machines become running.")
 	waitForNumMachinesInState(ctx, clusterv1.MachinePhaseRunning, waitForNumInput{
-		Client:    targetCluster.GetClient(),
-		Options:   []client.ListOption{client.InNamespace(namespace)},
+		Client:    input.TargetCluster.GetClient(),
+		Options:   []client.ListOption{client.InNamespace(input.Namespace)},
 		Replicas:  numberOfAllBmh,
-		Intervals: e2eConfig.GetIntervals(specName, "wait-machine-running"),
+		Intervals: input.E2EConfig.GetIntervals(input.SpecName, "wait-machine-running"),
 	})
 
 	By("PIVOTING TESTS PASSED!")
 }
 
-func installIronicBMO(targetCluster framework.ClusterProxy, isIronic, isBMO string) {
-	ironicTLSSetup := "IRONIC_TLS_SETUP"
-	ironicBasicAuth := "IRONIC_BASIC_AUTH"
+type installIronicBMOInput struct {
+	ManagementCluster          framework.ClusterProxy
+	BMPPath                    string
+	IsIronic                   bool
+	IsBMO                      bool
+	IronicTLSSetup             bool
+	IronicBasicAuth            bool
```

Comment:
since we are here, I would suggest renaming these to something more descriptive, i.e for someone new to the project:
```suggestion
	deployIronic             bool
	deployBMO                bool
	deployIronicTLSSetup     bool
	DeployIronicBasicAuth    bool
```

Reply:
thanks